src/js/app.js

```diff
@@ -0,0 +1,61 @@
+$(function() {
+  $('.limiter').hide();
+
+  $('form').on('submit', function(e){
+    e.preventDefault();
+    getResults($('#inputform').val());
+    $('#inputform').val('');
+    $('.container').hide();
+    $('.limiter').show();
+  })
+
+  $('#back-btn').click(function() {
+    location.reload();
+    console.log("reloaded");
+  })
+})
+
+function getResults(query) {
+  $.ajax({
+    type: "GET",
+    url: 'http://www.reddit.com/search.json?q=' + query,
+    dataType: "json",
+    success: function(reddit) {
+      $('#set-images').html(createImages(cleanse(reddit)));
+    }
+  })
+}
+
+function cleanse(ajaxObj) {
+  var cleanThumbs = [];
+  var childs = ajaxObj.data.children;
+  for (var i=0; i<ajaxObj.data.children.length; i++) {
+    if (childs[i].data.hasOwnProperty('preview')) {
+      console.log(childs[i].data.preview.length);
+      if (childs[i].data.preview.hasOwnProperty('images')) {
+        for (var j=0; j<childs[i].data.preview.images.length; j++) {
+          if (childs[i].data.preview.images[j].hasOwnProperty('source')) {
+            if (childs[i].data.preview.images[j].source.hasOwnProperty('url')) {
+              cleanThumbs.push(childs[i].data.preview.images[j].source.url);
+            }
+          }
+
+        }
+      }
+    }
+  }
```

Comment:
This is known as "nested statement hell". Once the course is over, or when you have time, I challenge you go back and **refactor** this. Some alternatives to think about:

* Use iterators (much more readable than `for` loops, and you won't need to go digging into `data` over and over again)
* Use functions (instead of all these nested statements, separate out the second `for` loop as a separate function)
* Use boolean expressions (nested `if` statements are not as readable and maintainable as using `&&`)